test/services/auth.test.ts

```diff
@@ -2,59 +2,128 @@
+
+        await createUser(userPayload);
+        const data = await authenticateLogin({
+          email: user.email,
+          password: user.password,
+        });
+
+        expect(data).not.toBeNull();
+        expect(data.email).toEqual(user.email);
+        expect(data.name).toEqual(user.name);
       });
+    });
 
-      expect(data).not.toBeNull();
-      expect(data.email).toEqual(user.email);
-      expect(data.name).toEqual(user.name);
+    describe('given there is NO user as credential', () => {
+      it('throws custom error', async () => {
+        await expect(
+          authenticateLogin({ email: 'dev@nimblehq.co', password: 'random' })
+        ).rejects.toHaveProperty('message', 'Provided email or password does not match.');
+      });
     });
-  });
 
-  describe('when there is no user as credential', () => {
-    it('throws bad request error.', async () => {
-      await expect(
-        authenticateLogin({ email: 'dev@nimblehq.co', password: 'random' })
-      ).rejects.toHaveProperty('code', 400);
+    describe('given NO credentials are passed', () => {
+      it('throws bad request error', async () => {
+        await expect(authenticateLogin({ email: '', password: '' })).rejects.toHaveProperty(
+          'message',
+          'Provided email or password does not match.'
+        );
+      });
     });
-  });
 
-  describe('when no credentials are passed', () => {
-    it('throws bad request error', async () => {
-      await expect(authenticateLogin({ email: '', password: '' })).rejects.toHaveProperty(
-        'code',
-        400
-      );
+    describe('given passed password is INCORRECT', () => {
+      it('throws custom error', async () => {
+        const userPayload = {
+          name: user.name,
+          email: 'user@nimblehq.co',
+          password: user.password,
+        };
+        await createUser(userPayload);
+
+        await expect(
+          authenticateLogin({
+            email: 'user@nimblehq.co',
+            password: '1234567',
+          })
+        ).rejects.toHaveProperty('message', 'Provided email or password does not match.');
+      });
     });
   });
 
-  describe('when passed password is incorrect', () => {
-    it('throws bad request error', async () => {
-      const userPayload = {
-        name: user.name,
-        email: 'user@nimblehq.co',
-        password: user.password,
-      };
-      await createUser(userPayload);
-
-      await expect(
-        authenticateLogin({
-          email: 'user@nimblehq.co',
-          password: '1234567',
-        })
-      ).rejects.toHaveProperty('code', 400);
+  describe('Authenticate google login', () => {
+    describe('given valid profile with no previous user', () => {
+      it('returns user after creating user and identity', async () => {
+        const userEmail = 'dev@nimblehq.co';
+        const profile = { email: userEmail, name: 'dev nimble' };
```

Comment:
As raised previously a test should follow the AAA pattern. So let's add a line break here after the Arrange block:

```suggestion
        const profile = { email: userEmail, name: 'dev nimble' };
        
```

Then the act block should be (one line):

```js
await authenticateGoogleLogin(profile);
```

Finally, the Assert block is:

```js
const authenticateUser = (await userRepository.findByEmail(userEmail)) as User;
const [identity] = await identityRepository.findByProvider(
  AuthProviders.google,
  authenticateUser.id
);

expect(identity).not.toBeNull();
expect(authenticateUser.email).toEqual(userEmail);
```

You need to recheck all tests added in this PR.

Reply:
Fixed in https://github.com/nimblehq/ic-nextjs-prabesh-dahal/pull/84/commits/f3a388edb96ff807b910abc71110e3dcca753c4b